docusaurus/docs/dev-docs/typescript/development/guides/documents-and-entries.md

```diff
@@ -0,0 +1,204 @@
+- **Generated Types:** Application types [have been generated](/dev-docs/typescript/development#generate-typings-for-content-types-schemas) and are accessible.
+
+## Type Imports
+
+The `UID` namespace contains literal unions representing the available resources in the application.
+
+```typescript
+import type { UID } from '@strapi/strapi';
+```
+
+- `UID.ContentType` represents a union of every content-type identifier in the application
+- `UID.Component` represents a union of every component identifier in the application
+- `UID.Schema` represents a union of every schema (content-type or component) identifier in the application 
+- And others...
+
+---
+
+Strapi provides a `Data` namespace containing several built-in types for entity representation.
+
+```typescript
+import type { Data } from '@strapi/strapi';
+```
+
+- `Data.ContentType` represents a Strapi document object
+- `Data.Component` represents a Strapi component object
+- `Data.Entity` represents either a document or a component
+
+:::tip
+Both the entities' type definitions and UIDs are based on the generated schema types for your application.
+
+In case of a mismatch or error, you can always [regenerate the types](/dev-docs/typescript/development#generate-typings-for-content-types-schemas). 
+:::
+
+## Usage
+
+### Generic Entities
+
+When dealing with generic data, it is recommended to use non-parametrized forms of the `Data` types.
+
+#### Generic Documents
+
+```typescript
+async function save(name: string, document: Data.ContentType) {
+  await writeCSV(name, document);
+  //                    ^ {
+  //                        id: Data.ID;
+  //                        documentId: string;
+  //                        createdAt?: DateTimeValue;
+  //                        updatedAt?: DateTimeValue;
+  //                        publishedAt?: DateTimeValue;
+  //                        ...
+  //                      }
+}
+```
+
+:::warning
+In the preceding example, the resolved properties for `document` are those common to every content-type.
+
+Other properties have to be checked manually using type guards.
+
+```typescript
+if ('my_prop' in document) {
+  return document.my_prop;
+}
+```
+:::
+
+#### Generic Components
+
+```typescript
+function renderComponent(parent: Node, component: Data.Component) {
+  const elements: Element[] = [];
+  const properties = Object.entries(component);
+
+  for (const [name, value] of properties) {
+    //        ^        ^
+    //        string   any
+    const paragraph = document.createElement('p');
+    
+    paragraph.textContent = `Key: ${name}, Value: ${value}`;
+    
+    elements.push(paragraph);
+  }
+  
+  parent.append(...elements);
+}
+```
+
+### Known Entities
+
+When manipulating known entities, it is possible to parametrize `Data` types for better type safety and intellisense.
```

Comment:
It does exist https://en.wikipedia.org/wiki/Parametrization

We mainly (always) use this spelling in the code/code documentation, thought it would be nice to stay consistent? :thinking:

Reply:
Let's keep it consistent then! Please feel free to ignore my suggestion :D